src/common/mfu_daos.c

```diff
@@ -319,7 +319,7 @@ int daos_parse_path(
         /* make sure there is enough memory allocated to concat basename */
         tmp = realloc(dattr.da_rel_path, path_len);
         if (tmp == NULL) {
-            rc = ENOMEM;
+            return -ENOMEM;
```

Comment:
@dsikich We would need to also make sure we free everything alloc'ed in this function before returning. Maybe we can add a `goto` to the cleanup at the bottom of this function?